### DISCUSSION TITLE
How to remove models with data in production

### DISCUSSION THREAD

**Discussion by reymon359** [prisma/prisma#17330] [Q&A]
Hey there 👋 

I want to delete 2 models from a postgreSQL production database that are no longer used but have data on them.

I am a bit afraid of removing them from the schema and running `prisma migrate dev` and then having issues with the migration generated in production. 

Therefore, my idea was to clean the database before:

- 1 make a query to the prod database to remove all the data from the tables related.
- 2 remove the models from the schema locally and run the migration.
- 3 push the migration to prod with the DB tables empty

What do you guys think? 

Should I do that or just don't be afraid and delete the models locally + run `prisma migrate dev` and then push the migration to prod?

Are there any other ways to accomplish what I want?

Thanks in advance!

**Comment by nurul3101** [comment]
Hello @reymon359 👋 

The approach you listed looks good. Other alternative would be to deleted the models locally and running `npx prisma migrate dev` command with [--create-only flag](https://www.prisma.io/docs/concepts/components/prisma-migrate/migrate-development-production#customizing-migrations) to just create the migration file without applying it to the database. You can inspect the migration file first and if it looks good to you then you can invoke `npx prisma migrate dev` again to apply it.

**Reply by reymon359** [reply]
Hey @nurul3101 

Thanks for the reply, I didn't know about that flag. It looks great for my needs, so thanks a lot for the suggestion, will give it a try when doing so 😄

**Reply by reymon359** [reply]
Coming back to say that the deletion was a success. Thanks for providing that flag. It was super useful 🙌

**Reply by nurul3101** [reply]
I am glad to hear that! 🙏 
Thank you so much for getting back and letting us know!

Please feel free to open a new discussion in case you face any issues. Also we now have an [support page](https://www.prisma.io/support) that has a universal search to quickly find resources across our Docs, Help Articles, Data Guide, etc. So feel free to use that as well 🙂

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
